TombEngine/Objects/Generic/Switches/airlock_switch.cpp

```diff
@@ -0,0 +1,82 @@
+#include "framework.h"
+#include "Game/control/control.h"
+#include "Specific/Input/Input.h"
+#include "Specific/level.h"
+#include "Game/Lara/lara.h"
+#include "Game/Lara/lara_helpers.h"
+#include "Objects/Generic/Switches/generic_switch.h"
+#include "Game/animation.h"
+#include "Game/collision/collide_item.h"
+#include "Game/items.h"
+
+using namespace TEN::Input;
+
+namespace TEN::Entities::Switches
+{
+	ObjectCollisionBounds LockSwitchBounds =
+	{
+		GameBoundingBox::Zero,
+		std::pair(
+			EulerAngles(ANGLE(-10.0f), ANGLE(-30.0f), ANGLE(-10.0f)),
+			EulerAngles(ANGLE(10.0f), ANGLE(30.0f), ANGLE(10.0f))
+		)
+	};
+
+	auto AirLockSwitchPos = Vector3i::Zero;
+
+	void AirLockSwitchCollision(short itemNumber, ItemInfo* laraItem, CollisionInfo* coll)
+	{
+		auto* laraInfo = GetLaraInfo(laraItem);
+		auto* switchItem = &g_Level.Items[itemNumber];
+
+		if (TrInput & IN_ACTION &&
+			switchItem->Animation.ActiveState == 0 &&
+			laraItem->Animation.ActiveState == LS_IDLE &&
+			laraItem->Animation.AnimNumber == LA_STAND_IDLE &&
+			laraInfo->Control.HandStatus == HandStatus::Free &&
+			switchItem->Status == ITEM_NOT_ACTIVE ||
+			laraInfo->Control.IsMoving && laraInfo->InteractedItem == itemNumber)
+		{
+			auto bounds = GameBoundingBox(switchItem);
+
+			LockSwitchBounds.BoundingBox.X1 = bounds.X1 - 256;
```

Comment:
Offsets can be demagicked with `BLOCK(0.25f)` and `BLOCK(0.5f)`.

Reply:
done